packages/ember-routing/lib/system/router.js

```diff
@@ -585,7 +585,7 @@ const EmberRouter = EmberObject.extend(Evented, {
       seen[name] = true;
 
       if (!handler) {
-        let DefaultRoute = routeOwner._lookupFactory('route:basic');
+        let DefaultRoute = routeOwner.factoryFor('route:basic').class;
```

Comment:
If we care about injections on `route:basic` being available on all generated routes, we can take the current implementation:

```js
let DefaultRoute = routeOwner._lookupFactory('route:basic');
routeOwner.register(fullRouteName, DefaultRoute.extend());
handler = routeOwner.lookup(fullRouteName);
// Later on may lookup this name again
```

And hack a version that should do the double-DI (DI for `route:basic` and `route:generated-thing` with the current implementation:

```js
let factory = routeOwner.factoryFor('route:basic');
let DefaultRouteProxy = {
  extend(...diArgs) {
    return {
      create(...createArgs) {
         return factory.create(assign({}, ...diArg, ...createArgs));
      }
    };
  },
  create(...createArgs) {
    /* Not sure this is ever going to be used in the current container double-extend implementation */
    return factory.create(assign({}, ...createArgs));
  }
}
```

However in 99.99999% of cases I don't expect someone is doing an injection on `route:basic` (or `controller:basic`), instead they are probably doing injections on the type. To keep these cases fast, we can check the injections list and if possible simple re-register the class:

```js
let DefaultRoute = routeOwner.factoryFor('route:basic').class;
routeOwner.register(fullRouteName, DefaultRoute);
handler = routeOwner.lookup(fullRouteName);
```

In this case DI on the type still works.